src/puppeteer_utils.js

```diff
@@ -59,20 +59,24 @@ const hideWebDriver = async (page) => {
  * Unlike Puppeteer's `addScriptTag` function, this function works on pages
  * with arbitrary Cross-Origin Resource Sharing (CORS) policies.
  *
- * Make sure that you're injecting the file after the page has loaded (after `page.goto()`). Otherwise,
- * the navigation will override the existing environment and the library will no longer be available.
- *
  * @param {Page} page
  *   Puppeteer <a href="https://pptr.dev/#?product=Puppeteer&show=api-class-page" target="_blank"><code>Page</code></a> object.
  * @param {String} filePath File path
+ * @param {Object} [options]
+ * @param {boolean} [options.surviveNavigations]
+ *   Enables the injected script to survive page navigations and reloads without need to be re-injected manually.
+ *   This does not mean, however, that internal state will be preserved. Just that it will be automatically
+ *   re-injected on each navigation before any other scripts get the chance to execute.
  * @return {Promise}
  * @memberOf puppeteer
  */
-const injectFile = async (page, filePath) => {
+const injectFile = async (page, filePath, options = {}) => {
     checkParamOrThrow(page, 'page', 'Object');
     checkParamOrThrow(filePath, 'filePath', 'String');
+    checkParamOrThrow(options, 'options', 'Object');
 
     const contents = await readFilePromised(filePath, 'utf8');
+    if (options.surviveNavigations) await page.evaluateOnNewDocument(contents);
```

Comment:
`page.evaluate()` and `page.evaluateOnNewDocument` could done done with `Promise.all()` concurrently

Reply:
Fair point